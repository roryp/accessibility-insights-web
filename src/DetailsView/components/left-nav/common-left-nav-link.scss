// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../../common/styles/colors.scss';
@import '../../../common/styles/common.scss';

:global(.is-selected) .left-nav-link-container {
    color: $always-black;
}

.left-nav-link-container {
    padding-left: 5%;
    padding-right: 8%;
    width: $detailsViewLeftNavWidth;
    color: $primary-text;
    display: flex;
    flex-wrap: nowrap;
    justify-content: flex-start;
    height: 100%;
    align-items: center;
    .link-icon {
        font-size: 24px;
    }
}
