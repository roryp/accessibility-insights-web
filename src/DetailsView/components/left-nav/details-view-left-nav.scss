// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../../common/styles/colors.scss';
@import '../../../common/styles/common.scss';

:global(.reflow-ui) {
    .left-nav {
        li {
            :global {
                .is-selected {
                    background-color: $nav-link-selected;
                    a,
                    button {
                        background-color: $nav-link-selected !important;
                    }
                }
                .is-expanded,
                .is-expanded + ul {
                    background-color: $nav-link-expanded;
                    .index-circle {
                        color: $neutral-0;
                        background: $index-circle-background;
                        border-color: $neutral-0;
                    }
                }
            }

            > div:hover {
                a,
                button {
                    background-color: $nav-link-hover;
                }
            }
        }
    }
}

:global(.details-view-body-nav-content-layout) {
    .left-nav {
        max-height: calc(100vh - (#{$detailsViewTotalHeaderHeight}));
        height: calc(100vh - (#{$detailsViewTotalHeaderHeight}));
    }

    &:global(.reflow-ui) .left-nav {
        max-height: calc(100vh - (#{$detailsViewHeaderBarHeight}));
        height: calc(100vh - (#{$detailsViewHeaderBarHeight}));
    }
}

.left-nav {
    overflow-y: auto;
    border-right: 1px solid $neutral-8;
    z-index: 100;
    background-color: $neutral-0;
    :global {
        .dark-gray {
            color: $neutral-60;
        }
        .ms-Nav-groupContent {
            margin-bottom: 0px;
        }
        .ms-Nav-link {
            height: fit-content;
        }
        .overview-label {
            padding-left: 10%;
            text-align: left;
            margin-top: 6px;
            margin-bottom: 6px;
            display: flex;
            flex-direction: column;
        }
        .overview-name {
            line-height: 20px;
        }
        .overview-percent {
            color: $neutral-55;
            line-height: 16px;
            font-size: 12px;
        }
        .index-circle {
            border: 1px solid;
            display: inline-block;
            border-radius: 50%;
            border: 1px solid;
            width: 1.5em;
            line-height: 1.5em;
            padding: 0.07em;
            margin-top: 0.23em;
        }
        .test-name {
            padding-left: 10%;
            display: inline-block;
        }
        .ms-Nav-navItems {
            margin-top: 0px !important;
        }
        .ms-Nav-compositeLink {
            a,
            button {
                border-left: $pivotItemLeftBorderWidth $pivotItemBorderStyle $neutral-0;
                border-top: 0px;
                border-bottom: 0px;
                text-align: center;
            }
            .ms-Button-label {
                color: $neutral-100;
            }
            .index-circle {
                color: $secondary-text;
                background: $neutral-0;
                border-color: $neutral-20;
            }
            &.is-selected {
                a,
                button {
                    background-color: $communication-tint-40;
                    border-left: $pivotItemLeftBorderWidth $pivotItemBorderStyle
                        $communication-primary;
                    .ms-Button-label {
                        color: $highlighted-text;
                    }
                    .overview-percent,
                    .dark-gray {
                        color: $overview-percent-selected;
                    }
                }
                a::after,
                button::after {
                    border-left: 0px;
                }
                .ms-Button-label {
                    color: $always-black;
                }
                .index-circle {
                    color: $neutral-0;
                    background: $index-circle-background;
                    border-color: $neutral-0;
                }
            }
        }
        .ms-Pivot-text {
            font-size: $fontSizeS;
        }
        .ms-Pivot-icon {
            font-size: 9px;
        }
        button {
            text-align: left;
        }
    }
}
