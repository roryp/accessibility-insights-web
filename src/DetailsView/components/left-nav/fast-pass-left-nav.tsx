// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { FeatureFlags } from 'common/feature-flags';
import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
import { INav } from 'office-ui-fabric-react';
import * as React from 'react';

import { NamedFC } from '../../../common/react/named-fc';
import { VisualizationType } from '../../../common/types/visualization-type';
import { NavLinkHandler } from './nav-link-handler';
import {
    VisualizationBasedLeftNav,
    VisualizationBasedLeftNavDeps,
} from './visualization-based-left-nav';

export type FastPassLeftNavDeps = {
    navLinkHandler: NavLinkHandler;
} & VisualizationBasedLeftNavDeps;
export type FastPassLeftNavProps = {
    deps: FastPassLeftNavDeps;
    selectedKey: string;
    featureFlagStoreData: FeatureFlagStoreData;
    onRightPanelContentSwitch: () => void;
    setNavComponentRef: (nav: INav) => void;
};

export const FastPassLeftNav = NamedFC<FastPassLeftNavProps>('FastPassLeftNav', props => {
    const { deps, setNavComponentRef } = props;

    const { navLinkHandler } = deps;

    const tests = [VisualizationType.Issues, VisualizationType.TabStops];

    if (props.featureFlagStoreData[FeatureFlags.needsReview]) {
        tests.push(VisualizationType.NeedsReview);
    }

    return (
        <VisualizationBasedLeftNav
            {...props}
            onLinkClick={navLinkHandler.onFastPassTestClick}
            visualizations={tests}
            onRightPanelContentSwitch={props.onRightPanelContentSwitch}
            setNavComponentRef={setNavComponentRef}
        />
    );
});
