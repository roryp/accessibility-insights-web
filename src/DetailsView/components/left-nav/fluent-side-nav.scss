// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../../common/styles/common.scss';

#side-nav-container {
    max-width: 25%;
    min-width: $reflowLeftNavWidth;
}

.left-nav-panel {
    :global(.ms-Panel-main) {
        width: $reflowLeftNavWidth;
    }
    :global(.ms-Panel-content) {
        padding: 0px;
        :global(.main-nav) {
            max-height: calc(100vh - (#{$detailsViewHeaderBarHeight}));
            height: calc(100vh - (#{$detailsViewHeaderBarHeight}));
        }
    }
}
