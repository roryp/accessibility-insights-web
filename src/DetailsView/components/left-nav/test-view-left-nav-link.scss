// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

.test-name {
    padding-left: 10%;
    display: inline-block;
}
