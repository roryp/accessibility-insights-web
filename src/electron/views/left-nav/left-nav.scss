// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../../common/styles/common.scss';

.left-nav {
    display: flex;
    flex-direction: column;
    min-width: $reflowLeftNavWidth;
    max-width: $reflowLeftNavWidth;
    border-right: $reflowLeftNavBorderWidth solid $neutral-alpha-8;
    background-color: $neutral-0;

    .header-container {
        display: flex;
        align-items: center;
        height: 40px;
        border-bottom: 1px solid $neutral-alpha-8;

        .rocket-icon {
            padding-left: 16px;
            padding-right: 16px;
            font-size: 18px;
        }

        h3 {
            font-size: 17px;
            font-weight: 600;
        }
    }
}
