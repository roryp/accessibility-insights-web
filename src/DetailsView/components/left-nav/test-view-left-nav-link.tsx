// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as styles from 'DetailsView/components/left-nav/common-left-nav-link.scss';
import * as React from 'react';
import { NamedFC } from '../../../common/react/named-fc';
import { BaseLeftNavLinkProps } from '../base-left-nav';

export const TestViewLeftNavLink = NamedFC<BaseLeftNavLinkProps>(
    'TestViewLeftNavLink',
    ({ link, renderIcon }) => {
        return (
            <span className={styles.leftNavLinkContainer} aria-hidden="true">
                {renderIcon(link)}
                <span className={'ms-Button-label test-name'}>{link.name}</span>
            </span>
        );
    },
);
