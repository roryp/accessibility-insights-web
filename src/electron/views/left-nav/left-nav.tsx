// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { NamedFC } from 'common/react/named-fc';
import { BaseLeftNav, BaseLeftNavLink } from 'DetailsView/components/base-left-nav';
import { NavLinkRenderer } from 'DetailsView/components/left-nav/nav-link-renderer';
import { LeftNavItem } from 'electron/types/left-nav-item';
import { LeftNavItemKey } from 'electron/types/left-nav-item-key';
import * as styles from 'electron/views/left-nav/left-nav.scss';
import { Icon } from 'office-ui-fabric-react';
import * as React from 'react';

export type LeftNavDeps = {
    leftNavItems: LeftNavItem[];
    navLinkRenderer: NavLinkRenderer;
};

export type LeftNavProps = {
    deps: LeftNavDeps;
    selectedKey: LeftNavItemKey;
};

export const leftNavAutomationId = 'left-nav';

export const LeftNav = NamedFC<LeftNavProps>('LeftNav', props => {
    const { deps } = props;
    const leftLinkItems: BaseLeftNavLink[] = deps.leftNavItems.map((item, index) => ({
        name: item.displayName,
        key: item.key,
        onClickNavLink: item.onSelect,
        onRenderNavLink: deps.navLinkRenderer.renderVisualizationLink,
        url: '',
        index: index + 1,
        iconProps: {
            className: 'hidden',
        },
        forceAnchor: true,
    }));

    return (
        <div data-automation-id={leftNavAutomationId} className={styles.leftNav}>
            <div className={styles.headerContainer}>
                <Icon iconName={'Rocket'} className={styles.rocketIcon} />
                <h3>FastPass</h3>
            </div>
            <BaseLeftNav selectedKey={props.selectedKey} links={leftLinkItems} />
        </div>
    );
});
