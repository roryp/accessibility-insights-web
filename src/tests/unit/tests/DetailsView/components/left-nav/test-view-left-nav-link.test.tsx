// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';

import {
    BaseLeftNavLink,
    BaseLeftNavLinkProps,
} from '../../../../../../DetailsView/components/base-left-nav';
import { TestViewLeftNavLink } from '../../../../../../DetailsView/components/left-nav/test-view-left-nav-link';

describe('TestViewLeftNavLink', () => {
    it('renders', () => {
        const props: BaseLeftNavLinkProps = {
            link: {
                name: 'test-props-link-name',
            } as BaseLeftNavLink,
            renderIcon: () => <i>test-props-renderIcon</i>,
        };

        const wrapper = shallow(<TestViewLeftNavLink {...props} />);

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
