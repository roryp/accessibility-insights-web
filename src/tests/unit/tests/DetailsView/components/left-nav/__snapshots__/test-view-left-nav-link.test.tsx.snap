// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestViewLeftNavLink renders 1`] = `
<span
  aria-hidden="true"
  className="leftNavLinkContainer"
  data-automation-id="test-view-left-nav-link"
>
  <i>
    test-props-renderIcon
  </i>
  <span
    className="testName"
  >
    test-props-link-name
  </span>
</span>
`;
